app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -114,16 +142,32 @@ export const moveStory = (
   destinationArray,
   sourceIndex,
   destinationIndex,
+  isSameColumn
 ) => {
   const newSourceArray = [...sourceArray];
   const [removed] = newSourceArray.splice(sourceIndex, 1);
-  const newDestinationArray = [...destinationArray];
-  const filteredArray = newDestinationArray.filter((el, index) => index !== sourceIndex);
-  filteredArray.splice(destinationIndex, 0, removed);
 
-  return [...filteredArray];
+  let newDestinationArray = [...destinationArray];
+  if (isSameColumn) {
+    newDestinationArray = newDestinationArray.filter((el, index) => index !== sourceIndex);
+  }
+  newDestinationArray.splice(destinationIndex, 0, removed);
+  const sortedStories = sortStories(newDestinationArray, destinationIndex, isSameColumn);
+
+  return [...sortedStories];
```

Comment:
Also, here you can avoid creating/ assigning to new variables by doing:
```js
const newDestionationArray = isSameColumn
  ? destinationArray.filter((_, index) => index !== sourceIndex).slice(destionationIndex, 0, removed)
  : destinationArray.slice(destionationIndex, 0, removed) 
  
return sortStories(newDestinationArray, destinationIndex, isSameColumn);
```

Unlike the `.splice` the `.slice` doesnt mutate the original array, so its safe to use like that :+1: